sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -65,7 +65,7 @@ case class CreateDataSourceTableCommand(
 
     var isExternal = true
     val optionsWithPath =
-      if (!new CaseInsensitiveMap(options).contains("path") && managedIfNoPath) {
+      if (!options.contains("path") && managedIfNoPath) {
```

Comment:
Maybe create a constant for the string "path" as a data source param? It occurs in quite a few places.